File: backend/app/Models/Stock.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Stock extends Model
{
    public $timestamps = false;

    protected $fillable = [
        'entrepot_id',
        'production_id',
        'produit',
        'quantite',
        'date_entree',
        'date_sortie',
        'seuil_alerte',
        'statut',
    ];

    public function entrepot()
    {
        return $this->belongsTo(Entrepot::class, 'entrepot_id');
    }

    public function production()
    {
        return $this->belongsTo(Production::class, 'production_id');
    }

    public function commandes()
    {
        return $this->hasMany(Commande::class, 'stock_id');
    }
}

File: mobile-app/lib/models/stock.dart
class Stock {
  final int id;
  final String produit;
  final double quantite;
  final double? seuilAlerte;
  final String? dateEntree;
  final String statut;
  final Map<String, dynamic>? entrepot;
  final Map<String, dynamic>? production;

  const Stock({
    required this.id,
    required this.produit,
    required this.quantite,
    this.seuilAlerte,
    this.dateEntree,
    required this.statut,
    this.entrepot,
    this.production,
  });

  bool get enAlerte =>
      seuilAlerte != null && quantite <= seuilAlerte!;

  factory Stock.fromJson(Map<String, dynamic> j) => Stock(
        id: j['id'],
        produit: j['produit'],
        quantite: (j['quantite'] as num).toDouble(),
        seuilAlerte: (j['seuil_alerte'] as num?)?.toDouble(),
        dateEntree: j['date_entree'],
        statut: j['statut'] ?? 'disponible',
        entrepot: j['entrepot'],
        production: j['production'],
      );
}
